### PR TITLE
Cache then network recipe

### PR DESCRIPTION
**Pull request by TimAbraldes**

### CODE REVIEW

File: cache-then-network/index.html [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  </form>
+
+  <form>Network:
+    <input id='network-fail' type='checkbox'>Fail</input>
+    Delay: <input id='network-delay' type='number' value='5000'></input>ms
+  </form>
+
+  <button id='getDataButton'>Get new data</button>
+
+  <script src="./sw.js"></script>
```
Comment (digitarald) [inline cache-then-network/index.html]: Why is the sw in the page?

**Comment by TimAbraldes** [comment]
Removed WIP tag. This no longer uses a ServiceWorker at all as that isn't necessary for this demo. It shows off the Cache and Fetch APIs from the main page. We should probably create a separate demo that does live-updating using a SW that communicates with the main page via `postMessage`

File: cache-then-network/index.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+          try {
+            handleCacheFetchCompletion(res);
+            resolve();
+          } catch (err) {
+            reject(err);
+          }
+        }, cacheDelay);
+      });
+    }).then(function() {
+      var now = window.performance.now();
```
Comment (digitarald) [inline cache-then-network/index.js]: If you use `Date.now()` you would not need to round here. Do you see sub-millisecond results?

File: cache-then-network/index.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    body {
+      font-family: monospace;
+      font-size: 1rem;
+      white-space: pre-wrap;
+    }
+  </style>
+</head>
+<body>
+
+  <label>Data: </label><output id='data'></output>
```
Comment (digitarald) [inline cache-then-network/index.html]: Consistency: " instead of ' (" use started in `head`)

File: cache-then-network/index.js (hunk 2 above)
Comment (TimAbraldes) [inline cache-then-network/index.js]: > Do you see sub-millisecond results? 

Not with `setTimeout` involved.

> If you use Date.now() you would not need to round here.

Changed!

File: cache-then-network/index.html (hunk 3 above)
Comment (TimAbraldes) [inline cache-then-network/index.html]: > Consistency: " instead of ' (" use started in head)

Changed all '"' to '''

**Pull request merged**